pkg/provider/azure_loadbalancer.go

```diff
@@ -2289,12 +2290,32 @@ func (az *Cloud) ensurePIPTagged(service *v1.Service, pip *network.PublicIPAddre
 	if serviceNames != nil {
 		configTags[consts.ServiceTagKey] = serviceNames
 	}
+
+	var systemTags []string
+	var systemTagsMap sets.String
+	if az.SystemTags != "" {
+		systemTags = strings.Split(az.SystemTags, consts.TagsDelimiter)
+		systemTagsMap = sets.NewString(systemTags...)
+	}
+
+	// if the systemTags is not set, just add/update new tags and not delete old tags
 	for k, v := range configTags {
 		if vv, ok := pip.Tags[k]; !ok || !strings.EqualFold(to.String(v), to.String(vv)) {
 			pip.Tags[k] = v
 			changed = true
 		}
 	}
+
+	// if the systemTags is set, delete the old tags
```

Comment:
add a function for this since it is referenced by multiple places

Reply:
done

Reply:
done